python/pynessie/client/nessie_client.py

```diff
@@ -75,13 +76,13 @@ def __init__(self: "NessieClient", config: confuse.Configuration) -> None:
         except confuse.exceptions.NotFoundError:
             self._base_branch = None
 
-    def list_references(self: "NessieClient") -> List[Reference]:
+    def list_references(self: "NessieClient") -> ReferencesResponse:
```

Comment:
Since this method does not take pagination parameters, would it make sense to iterate all pages inside it and return the full list (as before)?

Reply:
That's what it implicitly does, as the server doesn't support paging yet.

Reply:
If we have iteration in the client lib, then client will have have to update when the server starts supporting pagination.

Reply:
We do not have that for list-keys/tables either at the moment, probably because it's not supported on the server side. It's only implemented for the commit log.

Reply:
fair enough